### PR TITLE
VR > Cost/Buy Tickets

### PR DESCRIPTION
**Pull request by juanfra**
🎫 https://central.tri.be/issues/134145

🎥 https://www.loom.com/share/c014809c3ddd43a5be50afecb8d1adbd

### CODE REVIEW

File: tests/views_integration/Tribe/Events/Views/V2/Partials/Day_View/Event/CostTest.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		$event = tribe_events()->set_args(
+			[
+				'start_date' => '2018-01-01 10am',
+				'timezone'   => 'Europe/Paris',
+				'duration'   => 3 * HOUR_IN_SECONDS,
+				'title'      => 'Test Event - 2018-01-01 10am',
+				'status'     => 'publish',
+				'cost'       => '$25',
+			]
+		)->create();
```
Comment (bordoni) [inline tests/views_integration/Tribe/Events/Views/V2/Partials/Day_View/Event/CostTest.php]: Use `$this->get_mock_event()`

File: tests/views_integration/Tribe/Events/Views/V2/Partials/Day_View/Event/CostTest.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		$event = tribe_events()->set_args(
+			[
+				'start_date' => '2018-01-01 10am',
+				'timezone'   => 'Europe/Paris',
+				'duration'   => 3 * HOUR_IN_SECONDS,
+				'title'      => 'Test Event - 2018-01-01 10am',
+				'status'     => 'publish',
+				'recurring'  => true,
+			]
+		)->create();
```
Comment (bordoni) [inline tests/views_integration/Tribe/Events/Views/V2/Partials/Day_View/Event/CostTest.php]: Use `$this->get_mock_event()`

File: tests/views_integration/Tribe/Events/Views/V2/Partials/List_View/Event/CostTest.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		$event = tribe_events()->set_args(
+			[
+				'start_date' => '2018-01-01 10am',
+				'timezone'   => 'Europe/Paris',
+				'duration'   => 3 * HOUR_IN_SECONDS,
+				'title'      => 'Test Event - 2018-01-01 10am',
+				'status'     => 'publish',
+				'cost'       => '$25',
+			]
+		)->create();
```
Comment (bordoni) [inline tests/views_integration/Tribe/Events/Views/V2/Partials/List_View/Event/CostTest.php]: Use `$this->get_mock_event()`

File: tests/views_integration/Tribe/Events/Views/V2/Partials/List_View/Event/CostTest.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		$event = tribe_events()->set_args(
+			[
+				'start_date' => '2018-01-01 10am',
+				'timezone'   => 'Europe/Paris',
+				'duration'   => 3 * HOUR_IN_SECONDS,
+				'title'      => 'Test Event - 2018-01-01 10am',
+				'status'     => 'publish',
+				'recurring'  => true,
+			]
+		)->create();
```
Comment (bordoni) [inline tests/views_integration/Tribe/Events/Views/V2/Partials/List_View/Event/CostTest.php]: Use `$this->get_mock_event()`

File: src/resources/postcss/components/_small-cta.pcss [hunk 5]
```diff
@@ -6,13 +6,28 @@
...
+	}
+
 	.tribe-events-c-small-cta__link {
 		cursor: pointer;
 	}
 
-	.tribe-events-c-small-cta__price {
+	.tribe-events-c-small-cta__price,
+	.tribe-events-c-small-cta__stock {
 		@mixin desktop-body-3;
```
Comment (paulmskim) [inline src/resources/postcss/components/_small-cta.pcss]: you have this defined already on the markup using the `tribe-common-b3` class, we don't need this.

File: src/resources/postcss/components/_small-cta.pcss [hunk 6]
```diff
@@ -6,13 +6,28 @@
 
 .tribe-events {
 
+	.tribe-events-c-small-cta {
+
+		> * {
+			margin-left: 10px;
```
Comment (paulmskim) [inline src/resources/postcss/components/_small-cta.pcss]: is it possible to use `var(--spacer-1)` here?

File: src/resources/postcss/components/_small-cta.pcss [hunk 7]
```diff
@@ -6,13 +6,28 @@
...
-	.tribe-events-c-small-cta__price {
+	.tribe-events-c-small-cta__price,
+	.tribe-events-c-small-cta__stock {
 		@mixin desktop-body-3;
-		padding-left: 10px;
+	}
 
+	.tribe-events-c-small-cta__stock {
+		color: var(--color-text-secondary);
+		margin-left: 5px;
```
Comment (paulmskim) [inline src/resources/postcss/components/_small-cta.pcss]: is it possible to use `var(--spacer-0)` here?

File: src/resources/postcss/views/day/_event.pcss [hunk 8]
```diff
@@ -152,5 +152,11 @@
 		}
 	}
 
+	.tribe-events-calendar-day__event-cost {
+		margin-top: var(--spacer-2);
+		@media (--viewport-medium) {
```
Comment (paulmskim) [inline src/resources/postcss/views/day/_event.pcss]: space above `@media`

File: src/resources/postcss/views/list/_event.pcss [hunk 9]
```diff
@@ -205,4 +205,11 @@
 			visibility: visible;
 		}
 	}
+
+	.tribe-events-calendar-list__event-cost {
+		margin-top: var(--spacer-2);
+		@media (--viewport-medium) {
```
Comment (paulmskim) [inline src/resources/postcss/views/list/_event.pcss]: space above `@media`

File: tests/views_integration/Tribe/Events/Views/V2/Partials/Month_View/Calendar_Body/Day/Calendar_Events/Calendar_Event/Tooltip/CostTest.php [hunk 10]
```diff
@@ -4,10 +4,10 @@
 
 use Tribe\Test\Products\WPBrowser\Views\V2\HtmlPartialTestCase;
 
-class CtaTest extends HtmlPartialTestCase
+class CostTest extends HtmlPartialTestCase
 {
 
-	protected $partial_path = 'month/calendar-body/day/calendar-events/calendar-event/tooltip/cta';
+	protected $partial_path = 'month/calendar-body/day/calendar-events/calendar-event/tooltip/cost';
```
Comment (paulmskim) [inline tests/views_integration/Tribe/Events/Views/V2/Partials/Month_View/Calendar_Body/Day/Calendar_Events/Calendar_Event/Tooltip/CostTest.php]: fix tests here to use `$this->get_mock_event()`, also fix tests to match the test names and cases of the other tests (list and day view)

File: tests/views_integration/Tribe/Events/Views/V2/Partials/Month_View/Mobile_Events/Mobile_Day/Mobile_Event/CostTest.php [hunk 11]
```diff
@@ -3,10 +3,10 @@
 
 use Tribe\Test\Products\WPBrowser\Views\V2\HtmlPartialTestCase;
 
-class CtaTest extends HtmlPartialTestCase
+class CostTest extends HtmlPartialTestCase
 {
 
-	protected $partial_path = 'month/mobile-events/mobile-day/mobile-event/cta';
+	protected $partial_path = 'month/mobile-events/mobile-day/mobile-event/cost';
```
Comment (paulmskim) [inline tests/views_integration/Tribe/Events/Views/V2/Partials/Month_View/Mobile_Events/Mobile_Day/Mobile_Event/CostTest.php]: use `$this->get_mock_event()` here, also fix tests to match the test names and cases of the other tests (list and day view)

File: tests/views_integration/Tribe/Events/Views/V2/Partials/Day_View/Event/CostTest.php [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+		$this->assertMatchesSnapshot( $this->get_partial_html( [ 'event' => $event ] ) );
+	}
+
+	/**
+	 * Test render without cost
+	 */
+	public function test_render_without_cost() {
+		$event = $this->get_mock_event( 'events/single/1.template.json', [ 'ID' => 23, 'start_date' => '2018-01-01', 'end_date' => '2018-01-01' ] );
+		$event->cost = '$10';
```
Comment (paulmskim) [inline tests/views_integration/Tribe/Events/Views/V2/Partials/Day_View/Event/CostTest.php]: should not have cost in this test.

File: tests/views_integration/Tribe/Events/Views/V2/Partials/List_View/Event/CostTest.php [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+		$this->assertMatchesSnapshot( $this->get_partial_html( [ 'event' => $event ] ) );
+	}
+
+	/**
+	 * Test render without cost
+	 */
+	public function test_render_without_cost() {
+		$event = $this->get_mock_event( 'events/single/1.template.json', [ 'ID' => 23, 'start_date' => '2018-01-01', 'end_date' => '2018-01-01' ] );
+		$event->cost = '$10';
```
Comment (paulmskim) [inline tests/views_integration/Tribe/Events/Views/V2/Partials/List_View/Event/CostTest.php]: should not have cost in this test.

File: tests/views_integration/Tribe/Events/Views/V2/Partials/Month_View/Calendar_Body/Day/Calendar_Events/Calendar_Event/Tooltip/CostTest.php (hunk 10 above)
Comment (paulmskim) [reply tests/views_integration/Tribe/Events/Views/V2/Partials/Month_View/Calendar_Body/Day/Calendar_Events/Calendar_Event/Tooltip/CostTest.php]: still missing case without cost

File: tests/views_integration/Tribe/Events/Views/V2/Partials/Month_View/Mobile_Events/Mobile_Day/Mobile_Event/CostTest.php (hunk 11 above)
Comment (paulmskim) [reply tests/views_integration/Tribe/Events/Views/V2/Partials/Month_View/Mobile_Events/Mobile_Day/Mobile_Event/CostTest.php]: still missing test without cost

**Pull request merged**